<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>9 Interpreting model coefficients with ExploreModelMatrix | Statistical Analysis of Genome Scale Data 2023</title>
  <meta name="description" content="9 Interpreting model coefficients with ExploreModelMatrix | Statistical Analysis of Genome Scale Data 2023" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="9 Interpreting model coefficients with ExploreModelMatrix | Statistical Analysis of Genome Scale Data 2023" />
  <meta property="og:type" content="book" />
  
  
  <meta name="github-repo" content="lcolladotor/cshl_rstats_genome_scale_2023" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="9 Interpreting model coefficients with ExploreModelMatrix | Statistical Analysis of Genome Scale Data 2023" />
  
  
  

<meta name="author" content="Leonardo Collado-Torres" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  <link rel="shortcut icon" href="icon_32.png" type="image/x-icon" />
<link rel="prev" href="recount3-introduction.html"/>
<link rel="next" href="differential-gene-expression-analysis-with-limma.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a>
<ul>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#download-course-materials"><i class="fa fa-check"></i>Download course materials</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#code-of-conduct"><i class="fa fa-check"></i>Code of Conduct</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-schedule"><i class="fa fa-check"></i>Course Schedule</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#external-links"><i class="fa fa-check"></i>External links</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#course-prerequisites"><i class="fa fa-check"></i>Course Prerequisites</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#r-session-information"><i class="fa fa-check"></i>R session information</a></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html#license"><i class="fa fa-check"></i>License</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html"><i class="fa fa-check"></i><b>1</b> SPEAQeasy introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#overview-slides"><i class="fa fa-check"></i><b>1.1</b> 2022-04-20 overview slides</a></li>
<li class="chapter" data-level="1.2" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#speaqeasy-main-links"><i class="fa fa-check"></i><b>1.2</b> SPEAQeasy main links</a></li>
<li class="chapter" data-level="1.3" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#pipeline-outputs"><i class="fa fa-check"></i><b>1.3</b> Pipeline outputs</a></li>
<li class="chapter" data-level="1.4" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#exercises"><i class="fa fa-check"></i><b>1.4</b> Exercises</a></li>
<li class="chapter" data-level="1.5" data-path="speaqeasy-introduction.html"><a href="speaqeasy-introduction.html#solutions"><i class="fa fa-check"></i><b>1.5</b> Solutions</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html"><i class="fa fa-check"></i><b>2</b> smokingMouse study</a>
<ul>
<li class="chapter" data-level="2.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#introduction"><i class="fa fa-check"></i><b>2.1</b> Introduction</a></li>
<li class="chapter" data-level="2.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#overview-1"><i class="fa fa-check"></i><b>2.2</b> Overview</a></li>
<li class="chapter" data-level="2.3" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#goals"><i class="fa fa-check"></i><b>2.3</b> Goals</a>
<ul>
<li class="chapter" data-level="2.3.1" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#main-objective"><i class="fa fa-check"></i><b>2.3.1</b> Main objective</a></li>
<li class="chapter" data-level="2.3.2" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#secondary-objectives"><i class="fa fa-check"></i><b>2.3.2</b> Secondary objectives</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="smokingmouse-study.html"><a href="smokingmouse-study.html#study-design"><i class="fa fa-check"></i><b>2.4</b> Study design</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html"><i class="fa fa-check"></i><b>3</b> SummarizedExperiment review</a>
<ul>
<li class="chapter" data-level="3.1" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#overview-2"><i class="fa fa-check"></i><b>3.1</b> Overview</a></li>
<li class="chapter" data-level="3.2" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#quiz"><i class="fa fa-check"></i><b>3.2</b> Quiz</a></li>
<li class="chapter" data-level="3.3" data-path="summarizedexperiment-review.html"><a href="summarizedexperiment-review.html#exercises-1"><i class="fa fa-check"></i><b>3.3</b> Exercises</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html"><i class="fa fa-check"></i><b>4</b> smokingMouse RSE</a>
<ul>
<li class="chapter" data-level="4.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#download-data"><i class="fa fa-check"></i><b>4.1</b> Download data</a></li>
<li class="chapter" data-level="4.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#data-overview"><i class="fa fa-check"></i><b>4.2</b> Data overview</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#assays"><i class="fa fa-check"></i><b>4.2.1</b> Assays</a></li>
<li class="chapter" data-level="4.2.2" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#sample-data"><i class="fa fa-check"></i><b>4.2.2</b> Sample data</a></li>
<li class="chapter" data-level="4.2.3" data-path="smokingmouse-rse.html"><a href="smokingmouse-rse.html#gene-information"><i class="fa fa-check"></i><b>4.2.3</b> Gene Information</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="5" data-path="isee-introduction.html"><a href="isee-introduction.html"><i class="fa fa-check"></i><b>5</b> iSEE introduction</a>
<ul>
<li class="chapter" data-level="5.1" data-path="isee-introduction.html"><a href="isee-introduction.html#toy-rse-data"><i class="fa fa-check"></i><b>5.1</b> Toy RSE data</a></li>
<li class="chapter" data-level="5.2" data-path="isee-introduction.html"><a href="isee-introduction.html#isee"><i class="fa fa-check"></i><b>5.2</b> iSEE</a></li>
<li class="chapter" data-level="5.3" data-path="isee-introduction.html"><a href="isee-introduction.html#exercise-with-data-from-spatiallibd"><i class="fa fa-check"></i><b>5.3</b> Exercise with data from spatialLIBD</a></li>
<li class="chapter" data-level="5.4" data-path="isee-introduction.html"><a href="isee-introduction.html#community"><i class="fa fa-check"></i><b>5.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html"><i class="fa fa-check"></i><b>6</b> RSE plotting basics with smokingMouse</a>
<ul>
<li class="chapter" data-level="6.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#download-the-smokingmouse-data"><i class="fa fa-check"></i><b>6.1</b> Download the smokingMouse data</a></li>
<li class="chapter" data-level="6.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-preparation"><i class="fa fa-check"></i><b>6.2</b> Data preparation</a>
<ul>
<li class="chapter" data-level="6.2.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#data-normalization"><i class="fa fa-check"></i><b>6.2.1</b> Data normalization</a></li>
<li class="chapter" data-level="6.2.2" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#gene-filtering"><i class="fa fa-check"></i><b>6.2.2</b> Gene filtering</a></li>
</ul></li>
<li class="chapter" data-level="6.3" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#exploratory-data-analysis"><i class="fa fa-check"></i><b>6.3</b> Exploratory Data Analysis</a>
<ul>
<li class="chapter" data-level="6.3.1" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#quality-control-analysis"><i class="fa fa-check"></i><b>6.3.1</b> Quality Control Analysis</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="rse-plotting-basics-with-smokingmouse.html"><a href="rse-plotting-basics-with-smokingmouse.html#references"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html"><i class="fa fa-check"></i><b>7</b> biocthis introduction</a>
<ul>
<li class="chapter" data-level="7.1" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#related-past-materials"><i class="fa fa-check"></i><b>7.1</b> Related past materials</a></li>
<li class="chapter" data-level="7.2" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#biocthis-main-commands"><i class="fa fa-check"></i><b>7.2</b> biocthis main commands</a></li>
<li class="chapter" data-level="7.3" data-path="biocthis-introduction.html"><a href="biocthis-introduction.html#community-1"><i class="fa fa-check"></i><b>7.3</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="8" data-path="recount3-introduction.html"><a href="recount3-introduction.html"><i class="fa fa-check"></i><b>8</b> recount3 introduction</a>
<ul>
<li class="chapter" data-level="8.1" data-path="recount3-introduction.html"><a href="recount3-introduction.html#recount-projects"><i class="fa fa-check"></i><b>8.1</b> recount projects</a></li>
<li class="chapter" data-level="8.2" data-path="recount3-introduction.html"><a href="recount3-introduction.html#using-recount3"><i class="fa fa-check"></i><b>8.2</b> Using recount3</a></li>
<li class="chapter" data-level="8.3" data-path="recount3-introduction.html"><a href="recount3-introduction.html#exercise"><i class="fa fa-check"></i><b>8.3</b> Exercise</a></li>
<li class="chapter" data-level="8.4" data-path="recount3-introduction.html"><a href="recount3-introduction.html#community-2"><i class="fa fa-check"></i><b>8.4</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="9" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html"><i class="fa fa-check"></i><b>9</b> Interpreting model coefficients with ExploreModelMatrix</a>
<ul>
<li class="chapter" data-level="9.1" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r"><i class="fa fa-check"></i><b>9.1</b> Model objects in R</a></li>
<li class="chapter" data-level="9.2" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix"><i class="fa fa-check"></i><b>9.2</b> ExploreModelMatrix</a></li>
<li class="chapter" data-level="9.3" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1"><i class="fa fa-check"></i><b>9.3</b> Example 1</a></li>
<li class="chapter" data-level="9.4" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2"><i class="fa fa-check"></i><b>9.4</b> Example 2</a></li>
<li class="chapter" data-level="9.5" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3"><i class="fa fa-check"></i><b>9.5</b> Example 3</a></li>
<li class="chapter" data-level="9.6" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1"><i class="fa fa-check"></i><b>9.6</b> Exercise</a></li>
<li class="chapter" data-level="9.7" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more"><i class="fa fa-check"></i><b>9.7</b> To learn more</a></li>
<li class="chapter" data-level="9.8" data-path="interpreting-model-coefficients-with-exploremodelmatrix.html"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-3"><i class="fa fa-check"></i><b>9.8</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html"><i class="fa fa-check"></i><b>10</b> Differential gene expression analysis with limma</a>
<ul>
<li class="chapter" data-level="10.1" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#srp045638-data"><i class="fa fa-check"></i><b>10.1</b> SRP045638 data</a></li>
<li class="chapter" data-level="10.2" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#data-normalization-1"><i class="fa fa-check"></i><b>10.2</b> Data normalization</a></li>
<li class="chapter" data-level="10.3" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#differential-expression"><i class="fa fa-check"></i><b>10.3</b> Differential expression</a></li>
<li class="chapter" data-level="10.4" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#visualizing-degs"><i class="fa fa-check"></i><b>10.4</b> Visualizing DEGs</a></li>
<li class="chapter" data-level="10.5" data-path="differential-gene-expression-analysis-with-limma.html"><a href="differential-gene-expression-analysis-with-limma.html#community-4"><i class="fa fa-check"></i><b>10.5</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="11" data-path="differential-gene-expression-exercise-with-recount3-data.html"><a href="differential-gene-expression-exercise-with-recount3-data.html"><i class="fa fa-check"></i><b>11</b> Differential gene expression exercise with recount3 data</a>
<ul>
<li class="chapter" data-level="11.1" data-path="differential-gene-expression-exercise-with-recount3-data.html"><a href="differential-gene-expression-exercise-with-recount3-data.html#recap"><i class="fa fa-check"></i><b>11.1</b> Recap</a></li>
<li class="chapter" data-level="11.2" data-path="differential-gene-expression-exercise-with-recount3-data.html"><a href="differential-gene-expression-exercise-with-recount3-data.html#exercise-2"><i class="fa fa-check"></i><b>11.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="12" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html"><i class="fa fa-check"></i><b>12</b> DE model building with variancePartition</a>
<ul>
<li class="chapter" data-level="12.1" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#download-the-smokingmouse-data-and-subset-it"><i class="fa fa-check"></i><b>12.1</b> Download the smokingMouse data and subset it</a></li>
<li class="chapter" data-level="12.2" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#explore-gene-level-effects"><i class="fa fa-check"></i><b>12.2</b> Explore gene-level effects</a></li>
<li class="chapter" data-level="12.3" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#variance-partition-analysis"><i class="fa fa-check"></i><b>12.3</b> Variance Partition Analysis</a>
<ul>
<li class="chapter" data-level="12.3.1" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#canonical-correlation-analysis"><i class="fa fa-check"></i><b>12.3.1</b> Canonical Correlation Analysis</a></li>
<li class="chapter" data-level="12.3.2" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#fit-model-and-extract-fraction-of-variance-explained"><i class="fa fa-check"></i><b>12.3.2</b> Fit model and extract fraction of variance explained</a></li>
<li class="chapter" data-level="12.3.3" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#examine-expression-of-most-affected-genes-by-each-sample-variable"><i class="fa fa-check"></i><b>12.3.3</b> Examine expression of most affected genes by each sample variable</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="de-model-building-with-variancepartition.html"><a href="de-model-building-with-variancepartition.html#references-1"><i class="fa fa-check"></i>References</a></li>
</ul></li>
<li class="chapter" data-level="13" data-path="expression-heatmaps-with-complexheatmap.html"><a href="expression-heatmaps-with-complexheatmap.html"><i class="fa fa-check"></i><b>13</b> Expression heatmaps with ComplexHeatmap</a></li>
<li class="chapter" data-level="14" data-path="exercise-with-variancepartition-and-complexheatmap.html"><a href="exercise-with-variancepartition-and-complexheatmap.html"><i class="fa fa-check"></i><b>14</b> Exercise with variancePartition and ComplexHeatmap</a></li>
<li class="chapter" data-level="15" data-path="tbd.html"><a href="tbd.html"><i class="fa fa-check"></i><b>15</b> TBD</a>
<ul>
<li class="chapter" data-level="15.1" data-path="tbd.html"><a href="tbd.html#options"><i class="fa fa-check"></i><b>15.1</b> Options</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><i class="fa fa-check"></i><b>16</b> Introduction to the world of spatially-resolved transcriptomics with Visium data</a>
<ul>
<li class="chapter" data-level="16.1" data-path="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html"><a href="introduction-to-the-world-of-spatially-resolved-transcriptomics-with-visium-data.html#community-5"><i class="fa fa-check"></i><b>16.1</b> Community</a></li>
</ul></li>
<li class="chapter" data-level="17" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><i class="fa fa-check"></i><b>17</b> Downloading public Visium spatial data and visualizing it with spatialLIBD</a>
<ul>
<li class="chapter" data-level="17.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#download-data-from-10x-genomics"><i class="fa fa-check"></i><b>17.1</b> Download data from 10x Genomics</a></li>
<li class="chapter" data-level="17.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#add-gene-annotation-information"><i class="fa fa-check"></i><b>17.2</b> Add gene annotation information</a>
<ul>
<li class="chapter" data-level="17.2.1" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-10x"><i class="fa fa-check"></i><b>17.2.1</b> From 10x</a></li>
<li class="chapter" data-level="17.2.2" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#from-gencode"><i class="fa fa-check"></i><b>17.2.2</b> From Gencode</a></li>
</ul></li>
<li class="chapter" data-level="17.3" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#wrapper-functions-for-reading-the-data"><i class="fa fa-check"></i><b>17.3</b> Wrapper functions for reading the data</a></li>
<li class="chapter" data-level="17.4" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#visualize-spatial-plots"><i class="fa fa-check"></i><b>17.4</b> Visualize spatial plots</a></li>
<li class="chapter" data-level="17.5" data-path="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html"><a href="downloading-public-visium-spatial-data-and-visualizing-it-with-spatiallibd.html#exercise-3"><i class="fa fa-check"></i><b>17.5</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="18" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><i class="fa fa-check"></i><b>18</b> Re-use of bulk RNA-seq methods for spatial data exercise</a>
<ul>
<li class="chapter" data-level="18.1" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#spatial-registration"><i class="fa fa-check"></i><b>18.1</b> Spatial registration</a></li>
<li class="chapter" data-level="18.2" data-path="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html"><a href="re-use-of-bulk-rna-seq-methods-for-spatial-data-exercise.html#exercise-4"><i class="fa fa-check"></i><b>18.2</b> Exercise</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="final-r-session.html"><a href="final-r-session.html"><i class="fa fa-check"></i>Final R Session</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Statistical Analysis of Genome Scale Data 2023</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="interpreting-model-coefficients-with-exploremodelmatrix" class="section level1 hasAnchor" number="9">
<h1><span class="header-section-number">9</span> Interpreting model coefficients with ExploreModelMatrix<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#interpreting-model-coefficients-with-exploremodelmatrix" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Instructor: Leo</p>
<div id="model-objects-in-r" class="section level2 hasAnchor" number="9.1">
<h2><span class="header-section-number">9.1</span> Model objects in R<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#model-objects-in-r" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>Linear regression review <a href="https://lcolladotor.github.io/bioc_team_ds/helping-others.html#linear-regression-example" class="uri">https://lcolladotor.github.io/bioc_team_ds/helping-others.html#linear-regression-example</a></li>
<li>With R, we use the <code>model.matrix()</code> to build regression models using the <code>Y ~ X1 + X2</code> formula syntax as exemplified below.</li>
</ul>
<div class="sourceCode" id="cb62"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb62-1"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-1" aria-hidden="true" tabindex="-1"></a><span class="do">## ?model.matrix</span></span>
<span id="cb62-2"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-2" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> <span class="fu">with</span>(trees, <span class="fu">model.matrix</span>(<span class="fu">log</span>(Volume) <span class="sc">~</span> <span class="fu">log</span>(Height) <span class="sc">+</span> <span class="fu">log</span>(Girth)))</span>
<span id="cb62-3"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-3" aria-hidden="true" tabindex="-1"></a>mat</span>
<span id="cb62-4"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    (Intercept) log(Height) log(Girth)</span></span>
<span id="cb62-5"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1            1    4.248495   2.116256</span></span>
<span id="cb62-6"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2            1    4.174387   2.151762</span></span>
<span id="cb62-7"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3            1    4.143135   2.174752</span></span>
<span id="cb62-8"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4            1    4.276666   2.351375</span></span>
<span id="cb62-9"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5            1    4.394449   2.370244</span></span>
<span id="cb62-10"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6            1    4.418841   2.379546</span></span>
<span id="cb62-11"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7            1    4.189655   2.397895</span></span>
<span id="cb62-12"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8            1    4.317488   2.397895</span></span>
<span id="cb62-13"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 9            1    4.382027   2.406945</span></span>
<span id="cb62-14"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 10           1    4.317488   2.415914</span></span>
<span id="cb62-15"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 11           1    4.369448   2.424803</span></span>
<span id="cb62-16"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 12           1    4.330733   2.433613</span></span>
<span id="cb62-17"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 13           1    4.330733   2.433613</span></span>
<span id="cb62-18"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 14           1    4.234107   2.459589</span></span>
<span id="cb62-19"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 15           1    4.317488   2.484907</span></span>
<span id="cb62-20"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 16           1    4.304065   2.557227</span></span>
<span id="cb62-21"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  [ reached getOption(&quot;max.print&quot;) -- omitted 15 rows ]</span></span>
<span id="cb62-22"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; attr(,&quot;assign&quot;)</span></span>
<span id="cb62-23"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] 0 1 2</span></span>
<span id="cb62-24"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-24" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mat)</span>
<span id="cb62-25"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb62-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;(Intercept)&quot; &quot;log(Height)&quot; &quot;log(Girth)&quot;</span></span></code></pre></div>
<ul>
<li>How do we interpret the columns of our model matrix <code>mat</code>?</li>
</ul>
<div class="sourceCode" id="cb63"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb63-1"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(<span class="fu">lm</span>(<span class="fu">log</span>(Volume) <span class="sc">~</span> <span class="fu">log</span>(Height) <span class="sc">+</span> <span class="fu">log</span>(Girth), <span class="at">data =</span> trees))</span>
<span id="cb63-2"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-2" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb63-3"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-3" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb63-4"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; lm(formula = log(Volume) ~ log(Height) + log(Girth), data = trees)</span></span>
<span id="cb63-5"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb63-6"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Residuals:</span></span>
<span id="cb63-7"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       Min        1Q    Median        3Q       Max </span></span>
<span id="cb63-8"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.168561 -0.048488  0.002431  0.063637  0.129223 </span></span>
<span id="cb63-9"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb63-10"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb63-11"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error t value Pr(&gt;|t|)    </span></span>
<span id="cb63-12"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Intercept) -6.63162    0.79979  -8.292 5.06e-09 ***</span></span>
<span id="cb63-13"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; log(Height)  1.11712    0.20444   5.464 7.81e-06 ***</span></span>
<span id="cb63-14"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; log(Girth)   1.98265    0.07501  26.432  &lt; 2e-16 ***</span></span>
<span id="cb63-15"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb63-16"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb63-17"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb63-18"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Residual standard error: 0.08139 on 28 degrees of freedom</span></span>
<span id="cb63-19"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Multiple R-squared:  0.9777, Adjusted R-squared:  0.9761 </span></span>
<span id="cb63-20"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb63-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; F-statistic: 613.2 on 2 and 28 DF,  p-value: &lt; 2.2e-16</span></span></code></pre></div>
</div>
<div id="exploremodelmatrix" class="section level2 hasAnchor" number="9.2">
<h2><span class="header-section-number">9.2</span> ExploreModelMatrix<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exploremodelmatrix" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<ul>
<li>It’s a Bioconductor package which is useful to understand statistical models we use in differential expression analyses. It is interactive and helps us by creating some visual aids.
<ul>
<li><a href="http://www.bioconductor.org/packages/ExploreModelMatrix" class="uri">http://www.bioconductor.org/packages/ExploreModelMatrix</a></li>
</ul></li>
<li>For more details, check their paper <a href="https://doi.org/10.12688/f1000research.24187.2" class="uri">https://doi.org/10.12688/f1000research.24187.2</a>.</li>
<li>We’ll go over the examples they provide at <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html" class="uri">http://www.bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html</a></li>
</ul>
</div>
<div id="example-1" class="section level2 hasAnchor" number="9.3">
<h2><span class="header-section-number">9.3</span> Example 1<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb64"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb64-1"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="do">## Example data</span></span>
<span id="cb64-2"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-2" aria-hidden="true" tabindex="-1"></a>(sampleData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb64-3"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">genotype =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;A&quot;</span>, <span class="st">&quot;B&quot;</span>), <span class="at">each =</span> <span class="dv">4</span>),</span>
<span id="cb64-4"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">treatment =</span> <span class="fu">rep</span>(<span class="fu">c</span>(<span class="st">&quot;ctrl&quot;</span>, <span class="st">&quot;trt&quot;</span>), <span class="dv">4</span>)</span>
<span id="cb64-5"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-5" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb64-6"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   genotype treatment</span></span>
<span id="cb64-7"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 1        A      ctrl</span></span>
<span id="cb64-8"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 2        A       trt</span></span>
<span id="cb64-9"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 3        A      ctrl</span></span>
<span id="cb64-10"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 4        A       trt</span></span>
<span id="cb64-11"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 5        B      ctrl</span></span>
<span id="cb64-12"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 6        B       trt</span></span>
<span id="cb64-13"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 7        B      ctrl</span></span>
<span id="cb64-14"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 8        B       trt</span></span>
<span id="cb64-15"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-16"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-16" aria-hidden="true" tabindex="-1"></a><span class="do">## Let&#39;s make the visual aids provided by ExploreModelMatrix</span></span>
<span id="cb64-17"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-17" aria-hidden="true" tabindex="-1"></a>vd <span class="ot">&lt;-</span> ExploreModelMatrix<span class="sc">::</span><span class="fu">VisualizeDesign</span>(</span>
<span id="cb64-18"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">sampleData =</span> sampleData,</span>
<span id="cb64-19"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> genotype <span class="sc">+</span> treatment,</span>
<span id="cb64-20"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">textSizeFitted =</span> <span class="dv">4</span></span>
<span id="cb64-21"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-21" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb64-22"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-23"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-23" aria-hidden="true" tabindex="-1"></a><span class="do">## Now lets plot these images</span></span>
<span id="cb64-24"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb64-24" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(<span class="at">plotlist =</span> vd<span class="sc">$</span>plotlist)</span></code></pre></div>
<p><img src="_main_files/figure-html/EMM_example1-1.png" width="672" /></p>
<p>Interactively, we can run the following code:</p>
<div class="sourceCode" id="cb65"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb65-1"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="do">## We are using shiny again here</span></span>
<span id="cb65-2"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-2" aria-hidden="true" tabindex="-1"></a>app <span class="ot">&lt;-</span> <span class="fu">ExploreModelMatrix</span>(</span>
<span id="cb65-3"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">sampleData =</span> sampleData,</span>
<span id="cb65-4"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">designFormula =</span> <span class="sc">~</span> genotype <span class="sc">+</span> treatment</span>
<span id="cb65-5"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb65-6"><a href="interpreting-model-coefficients-with-exploremodelmatrix.html#cb65-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="fu">interactive</span>()) shiny<span class="sc">::</span><span class="fu">runApp</span>(app)</span></code></pre></div>
</div>
<div id="example-2" class="section level2 hasAnchor" number="9.4">
<h2><span class="header-section-number">9.4</span> Example 2<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-2" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html#example-2" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html#example-2</a></p>
</div>
<div id="example-3" class="section level2 hasAnchor" number="9.5">
<h2><span class="header-section-number">9.5</span> Example 3<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#example-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><a href="http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html#example-3" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/ExploreModelMatrix/inst/doc/ExploreModelMatrix.html#example-3</a></p>
</div>
<div id="exercise-1" class="section level2 hasAnchor" number="9.6">
<h2><span class="header-section-number">9.6</span> Exercise<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#exercise-1" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<style>
p.exercise  {
background-color: #E4EDE2;
padding: 9px;
border: 1px solid black;
border-radius: 10px;
font-family: sans-serif;
}
</style>
<p class="exercise">
<strong>Exercise 1</strong>:
Interpret <code>ResponseResistant.Treatmentpre</code> from the second example. It could be useful to take a screenshot and to draw some annotations on it.
</p>
<p class="exercise">
<strong>Exercise 2</strong>:
Whis is the <code>0</code> important at the beginning of the formula in the third example?
</p>
</div>
<div id="to-learn-more" class="section level2 hasAnchor" number="9.7">
<h2><span class="header-section-number">9.7</span> To learn more<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#to-learn-more" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p><em>A guide to creating design matrices for gene expression experiments</em>:</p>
<ul>
<li><a href="http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html" class="uri">http://bioconductor.org/packages/release/workflows/vignettes/RNAseq123/inst/doc/designmatrices.html</a></li>
<li><a href="https://f1000research.com/articles/9-1444" class="uri">https://f1000research.com/articles/9-1444</a></li>
</ul>
<p><em>“Model matrix not full rank”</em></p>
<ul>
<li><a href="http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#model-matrix-not-full-rank" class="uri">http://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#model-matrix-not-full-rank</a></li>
</ul>
</div>
<div id="community-3" class="section level2 hasAnchor" number="9.8">
<h2><span class="header-section-number">9.8</span> Community<a href="interpreting-model-coefficients-with-exploremodelmatrix.html#community-3" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Some of the <code>ExploreModelMatrix</code> authors:</p>
<ul>
<li><a href="https://twitter.com/CSoneson" class="uri">https://twitter.com/CSoneson</a></li>
<li><a href="https://twitter.com/FedeBioinfo" class="uri">https://twitter.com/FedeBioinfo</a></li>
<li><a href="https://twitter.com/mikelove" class="uri">https://twitter.com/mikelove</a></li>
</ul>

</div>
</div>
<div class="container">
    <footer class="site-footer">

      <center>
<p class="powered-by">
© 2011-2023. All thoughts and opinions here are my own. The icon was designed by <a href="https://www.linkedin.com/in/mauricio-guzman-6529b551/">Mauricio Guzmán</a> and is inspired by <a href="https://en.wikipedia.org/wiki/Huichol">Huichol culture</a>; it represents my community building interests.
</p>

<p class="powered-by copyright-license-text">
This work is licensed under <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">CC BY NC SA 4.0</a>
</p>

<p class="powered-by footer-license-icons">
    <a href="https://creativecommons.org/licenses/by-nc-sa/4.0" rel="noopener noreferrer" target="_blank">
      <img src="https://mirrors.creativecommons.org/presskit/icons/cc.svg" alt="CC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/by.svg" alt="CC by icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/nc.svg" alt="CC NC icon" width="32px">
      <img src="https://mirrors.creativecommons.org/presskit/icons/sa.svg" alt="CC SA icon" width="32px">
    </a>
</p>

<div id="clustrmaps-widget"></div><script type='text/javascript' id='clustrmaps' src='//cdn.clustrmaps.com/map_v2.js?cl=ffffff&w=300&t=m&d=Zo-deincaJGsqhqjIWsujtdoNOMJsnnZLefKmxDKVv8&co=124b73&cmo=dd53ff&cmn=4ec724'></script>

<p class="powered-by">
Published with <a href="https://bookdown.org/" target="_blank" rel="noopener">Bookdown</a>
</p>

</center>



<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=UA-45299226-1"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-45299226-1');
</script>

</footer>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="recount3-introduction.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-analysis-with-limma.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
</body>

</html>
